gnupg/gnupg.py

```diff
@@ -558,6 +590,37 @@ def _parse_keys(self, result):
             if keyword in valid_keywords:
                 getattr(result, keyword)(L)
 
+    def extend_key(self, keyid, validity='1y', passphrase=None, extend_subkey=True):
+        """Extends a GnuPG key by passing in new validity period (from now) through
+            subprocess's stdin
+
+        >>> import gnupg
+        >>> gpg = gnupg.GPG(homedir="doctests")
+        >>> key_input = gpg.gen_key_input()
+        >>> key = gpg.gen_key(key_input)
+        >>> gpg.extend_key(key.fingerprint, '2w', 'good passphrase')
+
+        :param str keyid: key shortID, longID, email_address or fingerprint
+        :param str validity: 0 or number of days (d), or weeks (*w) , or months (*m) or years (*y)
+                               to extend the key, from its creation date.
+        :param str passphrase: passphrase used when creating the key, leave None otherwise
+        :param bool extend_subkey: to indicate whether the first subkey will also extended
+                by the same period --default is True
```

Comment:
You're right! We should definitely act on all subkeys, but we're not sure how to determine how many keys there are. Any ideas how to do this?